src/book/mod.rs

```diff
@@ -332,25 +332,26 @@ fn determine_renderers(config: &Config) -> Vec<Box<Renderer>> {
 }
 
 /// Look at the `MDBook` and try to figure out what preprocessors to run.
-fn determine_preprocessors(config: &Config) -> Vec<Box<Preprocessor>> {
+fn determine_preprocessors(config: &Config) -> Result<Vec<Box<Preprocessor>>> {
+
+    let preprocess_list = match config.build.preprocess {
+        Some(ref p) => p,
+        // If no preprocessor field is set, default to the LinkPreprocessor. This allows you
+        // to disable the LinkPreprocessor by setting "preprocess" to an empty list.
+        None => return Ok(vec![Box::new(LinkPreprocessor::new())])
```

Comment:
I reckon we may want to pull this bit out into its own `default_preprocessors()` function, where we immediately return `vec![Box::new(LinkPreprocessor::new())]`.  I imagine that'll make things easier to update in the future, and it's immediately obvious what the default preprocessors are.

Reply:
Done - you're right, makes it a bit cleaner